hw/ip/rv_core_ibex/doc/boot-rom-patching.md

```diff
@@ -175,51 +175,68 @@ In this case, the dedicated SRAM can be locked after being loaded so that the pa
+    This makes the patch invalid and the patch loader will not enable it.
+  - `L` is the the lock bit.
+    When this is set, the system will disallow any writes to the 4 above-described registers that are used to configure the patched code region.
+- The second 32-bit word contains the remapping address base:
+  - `R_BASE` is the remapped address for the patched code region.
+    This is the SRAM address where the patch loader loads the code patch.
+    Once the patch loader enables a patch region, instruction fetches that match an address in the `[M_BASE..(M_BASE + P_SIZE - 1)]` interval will be replaced by the instruction at the same offset within the `[R_BASE..(R_BASE + P_SIZE - 1)]` instead.
+
+When applying ROM patches, the patch loader configures the Ibex instruction re-routing logic for each patch region, as follows:
+
+1. `IBUS_ADDR_MATCHING` is set to `(M_BASE & ~(P_SIZE - 1)) | ((P_SIZE - 1) >> 1)` in order for the Ibex remapping logic to form the ROM patch matching mask for that region.
+   See next section for more details.
+1. `IBUS_REMAP_ADDR` is set to `R_BASE`.
+1. `IBUS_ADDR_EN` is set to 1 if the patch loader evaluates the patch to be a valid one.
+1. `IBUS_REGWEN` is set to 0 if `L` is set to 1.
+
+#### Patch Match Logic Details
+
+As described in the previous section, the `IBUS_ADDR_MATCHING` register is configured by the patch loader with a value that aggregates both the ROM patch region base address and the region size.
+
+- If `M_BASE` value for the first entry in the OTP patch table is `yyy...yyy`, then the `IBUS_ADDR_MATCHING` register is configured as follows:
+
+| `IBUS_ADDR_MATCHING`| Code region size (in bytes) | `P_SIZE` |
+|---------------------|-----------------------------| ---------|
+| `yy yyyy yyyy yy01` | 4                           | 1        |
+| `yy yyyy yyyy y011` | 8                           | 2        |
+| `yy yyyy yyyy 0111` | 16                          | 4        |
+| `yy yyyy yyy0 1111` | 32                          | 8        |
 
-| `M_BASE`            | Size /bytes |
-|---------------------|-------------|
-| `yy yyyy yyyy yy01` | 4           |
-| `yy yyyy yyyy y011` | 8           |
-| `yy yyyy yyyy 0111` | 16          |
-| `yy yyyy yyy0 1111` | 32          |
 
-- A bitmask can be derived from this pattern as `mask = ~({M_BASE, 1} ^ ({M_BASE, 1} +1));`.
+- A bitmask can be derived from this pattern as `mask = (IBUS_ADDR_MATCHING ^ (IBUS_ADDR_MATCHING + 1));`.
   This mask will be `1` in all the positions with a `y` in the table.
   Then the redirection is defined by saying that it applies to an address `addr` if
 
 ```
-   (addr & mask) == ({M_BASE, 1'b0} & mask)
+   (addr & mask) == (IBUS_ADDR_MATCHING & mask)
 ```
 
-- If a redirection applies, the redirected address can be computed by adding the bottom bits of `addr` to `R_BASE`.
+- If a redirection applies, the redirected address can be computed by adding the bottom bits of `addr` to ``IBUS_REMAP_ADDR``.
```

Comment:
Nit: IBUS_REMAP_ADDR has become quoted twice here. It doesn't seem to affect rendering on github, though.